--- origsrc/lz4-r131/Makefile	2015-06-29 04:48:37.000000000 -0500
+++ src/lz4-r131/Makefile	2016-08-03 11:23:48.952298700 -0500
@@ -81,7 +81,7 @@ clean:
 
 #------------------------------------------------------------------------
 #make install is validated only for Linux, OSX, kFreeBSD and Hurd targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU))
+ifneq (,$(filter Linux Darwin GNU/kFreeBSD GNU CYGWIN%, $(shell uname)))
 
 install:
 	@cd $(LZ4DIR); $(MAKE) -e install
--- origsrc/lz4-r131/programs/Makefile	2015-06-29 04:48:37.000000000 -0500
+++ src/lz4-r131/programs/Makefile	2016-08-03 11:23:48.967323400 -0500
@@ -114,7 +114,7 @@ clean:
 
 #------------------------------------------------------------------------
 #make install is validated only for Linux, OSX, kFreeBSD and Hurd targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU))
+ifneq (,$(filter Linux Darwin GNU/kFreeBSD GNU CYGWIN%, $(shell uname)))
 
 install: lz4 lz4c
 	@echo Installing binaries
